lib/ui/bottombar/bottom_navigation_bar_controller.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:farmsmart_flutter/ui/bottombar/tab_navigator.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+class BottomNavigationBarController extends StatefulWidget {
```

Comment:
I've added the Controller bit because there's already a BottomNavigationBar name on flutter also I named this Controller because It controls the visibility of the tabs with the `Visibility` class. Maybe we can find another name for this

Reply:
Maybe "PersistentBottomNavigationBar" 
as it keeps the tabs states?

Reply:
Yeah! I like it ! Thanks! :)